tutorials/Python/neonUtilities/neonUtilitiesPython.md

```diff
@@ -0,0 +1,342 @@
+
+The data stacker package comes with a data file, table_types. The data file is needed for the package to work, and rpy2 doesn't load data by default. So we need to load it to the session and then pass it back to the R environment.
+
+First, load the data file:
+
+
+```python
+neonUdata = neonUtils.__rdata__
+table_types = neonUdata.fetch('table_types')['table_types']
+```
+
+Now, pass it back to the R environment:
+
+
+```python
+robjects.globalenv['table_types'] = table_types
+```
+
+Now run the `stackByTable()` function to stack the data. It requires two inputs: 
+1. The data product identifier (DPID) of the data you're stacking. DPIDs can be found in the <a href="http://data.neonscience.org/data-product-catalog" target="_blank">Data Product Catalog</a> and are in the form DP#.######.001
+2. A file path, the path to the zip file you downloaded from the NEON Data Portal.
+
+
+```python
+neonUtils.stackByTable('DP1.10100.001','/Downloads/NEON_isotope-soil-distrib-periodic.zip');
+```
+
+    Unpacked  NEON.D02.SCBI.DP1.10100.001.2014-08.expanded.20180308T180515Z.zip
+    
+    Unpacked  NEON.D07.ORNL.DP1.10100.001.2014-07.expanded.20180308T183441Z.zip
+    
+    Unpacked  NEON.D10.CPER.DP1.10100.001.2014-07.expanded.20180308T181823Z.zip
+    
+    Unpacked  NEON.D09.WOOD.DP1.10100.001.2014-08.expanded.20180308T182715Z.zip
+    
+    Unpacked  NEON.D10.STER.DP1.10100.001.2014-07.expanded.20180308T180231Z.zip
+    
+    Unpacked  NEON.D15.ONAQ.DP1.10100.001.2014-08.expanded.20180308T182837Z.zip
+    
+    Unpacked  NEON.D08.TALL.DP1.10100.001.2014-07.expanded.20180308T182745Z.zip
+    
+    Unpacked  NEON.D01.HARV.DP1.10100.001.2014-07.expanded.20180308T182420Z.zip
+    
+    Unpacked  NEON.D05.UNDE.DP1.10100.001.2014-07.expanded.20180308T180317Z.zip
+    
+    Unpacked  NEON.D01.BART.DP1.10100.001.2014-08.expanded.20180308T175912Z.zip
+    
+    Finished: All of the data are stacked into  1  tables!
+    
+    Copied the first available NEON.University_of_Wyoming_Stable_Isotope_Facility.bgc_CNiso_externalSummary.csv to /stackedFiles
+    Copied the first available variable definition file to /stackedFiles and renamed as variables.csv
+    Copied the first available validation file to /stackedFiles and renamed as validation.csv
+    Stacked  sls_soilStableIsotopes
+    
+
+
+Check the folder containing the original zip file from the Data Portal; you should now have a subfolder containing the unzipped and stacked files.
+
+## Download files to be stacked
+
+The function `zipsByProduct()` uses the <a href="http://data.neonscience.org/data-api" target="_blank">NEON API</a> to programmatically download data files for a given product. The files downloaded by `zipsByProduct()` can then be fed into `stackByTable()`.
+
+The function will create a new folder in the R working directory and write the files there. Set the working directory if it isn't set to where you want it.
```

Comment:
It isn't 100% clear to me if you are saying that the user needs to manually set the wd in R before the next step or if in the next step they will be setting the wd. I'm sure the later but clarification would be good.